library/Opus/SolrSearch/Searcher.php

```diff
@@ -34,231 +34,131 @@
-                    $this->getFacetSortsFromConfig());
-        }
-        else {
-            $params = array_merge($params,
-                    $this->getFacetLimitsFromConfig(),
-                    $this->getFacetSortsFromConfig());
-        }
-        return $params;
+            Opus_Log::get()->debug("query: " . $query->getQ());
+
+	        // get service adapter for searching
+	        $service = Opus_Search_Service::selectSearchingService( null, 'solr' );
+
+	        // basically create query
+	        $request = $service->createQuery()
+		        ->setFilter( new Opus_Search_Solr_Filter_Raw( $query->getQ() ) )
+		        ->setStart( $query->getStart() )
+		        ->setRows( $query->getRows() );
+
+
+	        switch ( $query->getSearchType() ) {
+		        case Opus_SolrSearch_Query::LATEST_DOCS :
+			        $request
+				        ->addSorting( $query->getSortField(), $query->getSortOrder() );
+
+		        case Opus_SolrSearch_Query::DOC_ID :
+					if ( $query->isReturnIdsOnly() ) {
+						$request
+							->setFields( 'id' );
+					} else {
+						$request
+							->setFields( array( '*', 'score' ) );
+					}
+			        break;
+
+		        case Opus_SolrSearch_Query::FACET_ONLY :
+					$facet = Opus_Search_Facet_Set::create()
+						->setFacetOnly();
+
+					$facet->addField( $query->getFacetField() )
+						->setMinCount( 1 )
+						->setLimit( -1 );
+
+					$request->setFacet( $facet );
+		            break;
+
+		        default :
+					$request->addSorting( $query->getSortField(), $query->getSortOrder() );
+
+			        if ( $query->isReturnIdsOnly() ) {
+				        $request
+					        ->setFields( 'id' );
+			        } else {
+				        $request
+					        ->setFields( array( '*', 'score' ) );
+
+				        $facet = Opus_Search_Facet_Set::create();
+
+				        if ( isset( $this->facetArray ) ) {
+					        $facet->overrideLimits( $this->facetArray );
+				        }
+
+		                $fields = Opus_Search_Config::getFacetFields( $facet->getSetName(), 'solr' );
+				        if ( empty( $fields ) ) {
+					        // no facets are being configured
+					        Opus_Log::get()->warn("Key searchengine.solr.facets is not present in config. No facets will be displayed.");
+				        } else {
+					        $request->setFacet( $facet->setFields( $fields ) );
+				        }
+	                }
+
+
+			        $fq = $query->getFilterQueries();
+			        if ( !empty( $fq ) ) {
+				        foreach ( $fq as $index => $sub ) {
+					        $request->setSubFilter( "fq$index", new Opus_Search_Solr_Filter_Raw( $sub ) );
+				        }
+			        }
+	        }
+
+	        $response = $service->customSearch( $request );
+
+	        if ( $validateDocIds ) {
+		        $response->dropLocallyMissingMatches();
+	        }
+
+	        return $response;
+        }
+        catch ( Opus_Search_InvalidServiceException $e ) {
+	        return $this->mapException( Opus_SolrSearch_Exception::INVALID_QUERY, $e );
```

Comment:
I think the error codes are switched, INVALID_QUERY for Opus_Search_InvalidServiceException and vice versa in the next catch.